UnitTests/CacheTest/LocalDataGatewayTest/UserLocalDataGatewayTest.cs

```diff
@@ -0,0 +1,281 @@
+using mRides_app.Cache.MemoryCache;
+using NUnit.Framework;
+using System.Reflection;
+using System.Collections.Generic;
+using System;
+using Android.Graphics;
+using System.Net;
+using mRides_app.Cache.LocalDatabase;
+using mRides_app.Cache.LocalDataGateways;
+using SQLite;
+using mRides_app.Models;
+
+namespace UnitTests
+{
+    [TestFixture]
+    public class UserLocalDataGatewayTest
+    {
+        [SetUp]
+        public void Setup()
+        {
+            // Before each test, clear the database
+            var docsFolder = System.Environment.GetFolderPath(System.Environment.SpecialFolder.MyDocuments);
+            var path = System.IO.Path.Combine(docsFolder, MRidesLocalDatabase.MRIDES_LOCAL_DB_FILENAME);
+            System.IO.File.Delete(path);
+
+            // Clear singleton
+            SetStaticField(typeof(MRidesLocalDatabase), "instance", null);
+            SetStaticField(typeof(UserLocalDataGateway), "instance", null);
+        }
+
+
+        [TearDown]
+        public void Tear()
+        {
+
+        }
+
+        // -------------------------------------------------------------------------------------------------------------
+        // UNIT TESTS
+        // -------------------------------------------------------------------------------------------------------------
+        
+        [Test]
+        public void AddUser()
+        {
+            // Id of the user used for testing
+            int userId = 12345;
+            
+            // Create a mock user to be stored in the database
+            User user = new User
+            {
+                id = userId,
+                facebookID = userId,
```

Comment:
Maybe you should use two different ids for userId and facebookID in your tests to make sure that it saves the right thing for both